layers/generated/categories.cpp

```diff
@@ -0,0 +1,2103 @@
+/* *** THIS FILE IS GENERATED - DO NOT EDIT! ***
+ * See parameter_validation_generator.py for modifications
+ *
+ * Copyright (c) 2023 The Khronos Group Inc.
+ * Copyright (c) 2023 LunarG, Inc.
+ * Copyright (C) 2023 Google Inc.
+ * Copyright (c) 2023 Valve Corporation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * Author: Mark Lobodzinski <mark@LunarG.com>
+ * Author: Dave Houlton <daveh@LunarG.com>
+ */
+
+
+#include "chassis.h"
+#include "stateless/stateless_validation.h"
+
+
+template<>
+std::vector<VkPipelineCacheHeaderVersion> ValidationObject::ValidParamValues() const {
+    // TODO (ncesario) This is not ideal as we compute the enabled extensions every time this function is called.
+    //      Ideally "values" would be something like a static variable that is built once and this function returns
+    //      a span of the container. This does not work for applications which create and destroy many instances and
+    //      devices over the lifespan of the project (e.g., VLT).
+    constexpr std::array CoreVkPipelineCacheHeaderVersionEnums = { VK_PIPELINE_CACHE_HEADER_VERSION_ONE,  };
+    static const layer_data::unordered_map<const ExtEnabled DeviceExtensions::*, std::vector<VkPipelineCacheHeaderVersion>> ExtendedVkPipelineCacheHeaderVersionEnums = {
+    };
+    std::vector<VkPipelineCacheHeaderVersion> values(CoreVkPipelineCacheHeaderVersionEnums.cbegin(), CoreVkPipelineCacheHeaderVersionEnums.cend());
+    std::set<VkPipelineCacheHeaderVersion> unique_exts;
+    for (const auto& itr : ExtendedVkPipelineCacheHeaderVersionEnums) {
+        if (IsExtEnabled(device_extensions.*itr.first)) {
+            unique_exts.insert(itr.second.cbegin(), itr.second.cend());
+        }
+    }
```

Comment:
Use C++17 structured bindings

```suggestion
    for (const auto& [extension, enums] : ExtendedVkPipelineCacheHeaderVersionEnums) {
        if (IsExtEnabled(device_extensions.*extension)) {
            unique_exts.insert(enums.cbegin(), enums.cend());
        }
    }
```